maths/area_under_curve.py

```diff
@@ -3,58 +3,81 @@
 """
 from __future__ import annotations
 
-from collections.abc import Callable
+
+def evaluate(arr: list[float], x: float) -> float:
+    """
+    This function takes a list of coefficients
+    and a value of x as inputs and returns
+    the value of the polynomial at that point.
+    :param arr: a list containing coefficients
+    :param x: value of x
+
+    >>> arr = [5]
+    >>> f"{evaluate(arr,2)}"
+    '5.0'
+    >>> arr = [5,1]
+    >>> f"{evaluate(arr,2)}"
+    '11.0'
+    """
+    y = 0.0
+    for i in range(len(arr)):
+        y += arr[i] * (x ** (len(arr) - i - 1))
+    return y
 
 
 def trapezoidal_area(
-    fnc: Callable[[float], float],
+    fnc: list[float],
     x_start: float,
     x_end: float,
     steps: int = 100,
 ) -> float:
     """
     Treats curve as a collection of linear lines and sums the area of the
     trapezium shape they form
-    :param fnc: a function which defines a curve
+    :param fnc: a list which defines a curve of polynomial
     :param x_start: left end point to indicate the start of line segment
     :param x_end: right end point to indicate end of line segment
     :param steps: an accuracy gauge; more steps increases the accuracy
     :return: a float representing the length of the curve
 
-    >>> def f(x):
-    ...    return 5
+    >>> f = [5.0]
     >>> f"{trapezoidal_area(f, 12.0, 14.0, 1000):.3f}"
     '10.000'
-    >>> def f(x):
-    ...    return 9*x**2
+    >>> f = [9.0,0.0,0.0]
     >>> f"{trapezoidal_area(f, -4.0, 0, 10000):.4f}"
     '192.0000'
     >>> f"{trapezoidal_area(f, -4.0, 4.0, 10000):.4f}"
     '384.0000'
     """
-    x1 = x_start
-    fx1 = fnc(x_start)
+    dx = (x_end - x_start) / steps
     area = 0.0
-    for _ in range(steps):
-        # Approximates small segments of curve as linear and solve
-        # for trapezoidal area
-        x2 = (x_end - x_start) / steps + x1
-        fx2 = fnc(x2)
-        area += abs(fx2 + fx1) * (x2 - x1) / 2
-        # Increment step
-        x1 = x2
-        fx1 = fx2
+    for i in range(steps):
+        x1 = x_start + i * dx
+        x2 = x_start + (i + 1) * dx
+        fx1 = evaluate(fnc, x1)
+        fx2 = evaluate(fnc, x2)
+        area += abs((fx2 + fx1) * dx / 2)
     return area
 
 
 if __name__ == "__main__":
+    m = int(input("Enter order of polynomial: ").strip())
```

Comment:
Instead of all user inputs and other things, only below will be enough.        
  `
          if __name__ == "__main__":
              import doctest
              doctest.testmod()
          `